variables.tf

```diff
@@ -290,6 +290,12 @@ variable "container_filter" {
   }
 }
 
+variable "prometheus_config" {
+  description = "Prometheus configuration for the agent. If you want to enable Prometheus configuration provide the prometheus.yaml file content in `hcl` format. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-monitoring-agent/blob/main/solutions/fully-configurable/DA-types.md#prometheus)."
+  type        = map(any)
+  default     = {}
```

Comment:
I think you should set `nullable = false` because the length check you are doing wil probably fail if user passed `null` for this?